CONTRIBUTING.md

```diff
@@ -122,6 +122,22 @@ branch. Steps:
    git push -u origin my-feature
    ```
 
+## Local Developer Setup
+1. Make sure you have a compatible version of Python 3 installed
+2. From the project root, create a virtual environment and install all development dependencies
+    ```bash
+   # Create virtual environment
+   python3 -m venv venv
+   
+   # Activate the virtual environment
+   source venv/bin/activate
```

Comment:
This is good, but I wonder if we should put some thought into telling people what to do if they're using cmd or powershell too.  Is this the time?

Reply:
Good idea.  I added instructions for cmd and powershell.